### ISSUE TITLE
FNX3-16808 ⁃ [Bug] Opening custom tab in the browser flashes previous session

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#3131]
## Steps to reproduce
-Open Gmail or LinkedIn and tap on a link to open in a custom tab (linked opened in twitter or whatsapp don't open in a custom tab)
-Once the custom tab is open, tap on the 3 dot menu, then tap on Open in Firefox Preview

### Expected behavior
-Page opens in Firefox Preview

### Actual behavior
-The most recent open tab in Firefox Preview opens

### Device information
* Android device: Pixel 3, Android 9
* Fenix version: 1.0.1923

**Comment by sblatz** [comment]
I'm not able to reproduce this on my Galaxy S9. That being said, there is a brief flash of the previous tab. I believe this is because we're now restoring the session (which takes a minute) rather than just launching the app with the URL. See: https://github.com/mozilla-mobile/fenix/pull/2796

![working 2019-06-04 14_43_57](https://user-images.githubusercontent.com/4400286/58915976-33672880-86d7-11e9-9961-752244d00503.gif)

**Comment by vesta0** [comment]
I think @colintheshots fixed the original issue (I couldn't see it since the nightly build was broken). 

@sblatz what can we do about the flash?

**Comment by sblatz** [comment]
I think the problem is that the current session doesn't switch over immediately. We would want to look into a solution that increases the animation length, displays a "blank" page, or something similar.

**Comment by sblatz** [comment]
I've spent some time investigating this and can't seem to figure it out. Perhaps someone else can continue looking at it. 

Somehow Sample Browser and Reference Browser are able to avoid this, so I'm sure a solution exists. That being said, even when the first line of code I executed was to go to Home Fragment (inside of `handleOpenedFromExternalSourceIfNecessary`) we'd still see a flash of the browser fragment, as the `IntentReceiverActivity` is already beginning the loading of the new browser.
